lighthouse-core/fraggle-rock/api.js

```diff
@@ -18,9 +20,48 @@ async function startFlow(page, options) {
   return new UserFlow(page, options);
 }
 
+/**
+ * @param  {Parameters<navigationGather>} params
+ * @return {Promise<LH.RunnerResult|undefined>}
+ */
+async function navigation(...params) {
+  const gatherResult = await navigationGather(...params);
+  return Runner.audit(gatherResult.artifacts, gatherResult.runnerOptions);
+}
+
+/**
+ * @param  {Parameters<snapshotGather>} params
+ * @return {Promise<LH.RunnerResult|undefined>}
+ */
+async function snapshot(...params) {
+  const gatherResult = await snapshotGather(...params);
+  return Runner.audit(gatherResult.artifacts, gatherResult.runnerOptions);
+}
+
+/**
+ * @param  {Parameters<startTimespanGather>} params
+ * @return {Promise<{endTimespan: () => Promise<LH.RunnerResult|undefined>}>}
+ */
+async function startTimespan(...params) {
+  const {endTimespanGather} = await startTimespanGather(...params);
+  const endTimespan = async () => {
+    const gatherResult = await endTimespanGather();
+    return Runner.audit(gatherResult.artifacts, gatherResult.runnerOptions);
+  };
+  return {endTimespan};
+}
+
+/**
+ * @param {LH.FlowResult} flowResult
+ */
+async function generateFlowReport(flowResult) {
```

Comment:
I added this api function for 2 reasons:
- Allows us to generate the flow result once when we generate flow fixtures with `--view` flag
- We will eventually need a way to generate a flow report for a result object generated from artifacts and not the `UserFlow` class